args.go

```diff
@@ -544,13 +544,31 @@ func (s *argsScanner) next(kv *argsKV) bool {
 }
 
 func decodeArgAppend(dst, src []byte) []byte {
-	if bytes.IndexByte(src, '%') < 0 && bytes.IndexByte(src, '+') < 0 {
+	idxPercent := bytes.IndexByte(src, '%')
+	idxPlus := bytes.IndexByte(src, '+')
+	if idxPercent < 0 && idxPlus < 0 {
 		// fast path: src doesn't contain encoded chars
 		return append(dst, src...)
 	}
 
+	idx := 0
```

Comment:
i dont think   it  is  graceful. better solution is  welcome

Reply:
I would do this, which has less comparisons:
```go
idx := 0
if idxPercent == -1 {
    if idxPlus != -1 {
        idx = idxPlus
    }
} else if idxPlus == -1 {
    idx = idxPercent
} else if idxPercent > idxPlus {
    idx = idxPlus
} else {
    idx = idxPercent
}
```

Reply:
thx，it  be  better. i  will  change it.
but the block
```
if idxPlus != -1 {   
     idx = idxPlus   
} 
```
is  unnecessary.    it is  not  possible the  `idxPercent `  and `idxPlus`  both  are  -1

Reply:
You could combine it with the check above to do less comparisons.

Reply:
> You could combine it with the check above to do less comparisons.

thanks @erikdubbelboer